src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseSpanBasedStringConcatTests.cs

```diff
@@ -0,0 +1,829 @@
+                    @"String.Concat(foo, 3.14159.ToString(), bar.AsSpan(1))"
+                };
+                yield return new[]
+                {
+                    @"3.14159 & foo.Substring(1) & 6.02e23",
+                    @"String.Concat(3.14159.ToString(), foo.AsSpan(1), 6.02e23.ToString())"
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Data_NonStringOperands_VB))]
+        public Task NonStringOperands_AreConvertedToString_VB(string testExpression, string fixedExpression)
+        {
+            string format = @"
+Dim count As Integer = 11
+Dim thing As Object = New Object()
+Dim maybe As TimeSpan? = Nothing
+Dim s = {0}";
+            var culture = CultureInfo.InvariantCulture;
+
+            var test = new VerifyVB.Test
+            {
+                TestCode = VBUsings + VBWithBody(string.Format(culture, format, $"{{|#0:{testExpression}|}}")),
+                FixedCode = VBUsings + VBWithBody(string.Format(culture, format, fixedExpression)),
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics = { VerifyVB.Diagnostic(Rule).WithLocation(0) }
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(@"foo.Substring(1) + (string)explicitTo", @"string.Concat(foo.AsSpan(1), (string)explicitTo)")]
+        [InlineData(@"(string)explicitTo + foo.Substring(1)", @"string.Concat((string)explicitTo, foo.AsSpan(1))")]
+        [InlineData(@"foo.Substring(1) + (string)thing", @"string.Concat(foo.AsSpan(1), (string)thing)")]
+        [InlineData(@"(string)thing + foo.Substring(1)", @"string.Concat((string)thing, foo.AsSpan(1))")]
+        [InlineData(@"foo.Substring(1) + (thing as string)", @"string.Concat(foo.AsSpan(1), thing as string)")]
+        [InlineData(@"(thing as string) + foo.Substring(1)", @"string.Concat(thing as string, foo.AsSpan(1))")]
+        public Task ExplicitConversions_ArePreserved_CS(string testExpression, string fixedExpression)
+        {
+            string helperTypes = @"
+public class ExplicitTo
+{
+    public static explicit operator string(ExplicitTo operand) => operand?.ToString();
+}
+
+public class ExplicitFrom
+{
+    public static explicit operator ExplicitFrom(string operand) => new ExplicitFrom();
+}";
+            string format = @"
+var explicitTo = new ExplicitTo();
+object thing = bar;
+var s = {0};";
+            var culture = CultureInfo.InvariantCulture;
+            string testStatements = string.Format(culture, format, $@"{{|#0:{testExpression}|}}");
+            string fixedStatements = string.Format(culture, format, fixedExpression);
+
+            var test = new VerifyCS.Test
+            {
+                TestState = { Sources = { CSUsings + CSWithBody(testStatements), helperTypes } },
+                FixedState = { Sources = { CSUsings + CSWithBody(fixedStatements), helperTypes } },
```

Comment:
Was it intentional to use TestState and FixedState instead of TestCode and FixedCode in this specific test?

Reply:
Yes, because I defined some helper types in a separate source file.

Reply:
@NewellClark Good. I didn't notice that.